nifi-nar-bundles/nifi-kudu-bundle/nifi-kudu-processors/src/main/java/org/apache/nifi/processors/kudu/PutKudu.java

```diff
@@ -307,9 +302,11 @@ private void trigger(final ProcessContext context, final ProcessSession session,
         final List<RowError> pendingRowErrors = new ArrayList<>();
         for (FlowFile flowFile : flowFiles) {
             try (final InputStream in = session.read(flowFile);
-                 final RecordReader recordReader = recordReaderFactory.createRecordReader(flowFile, in, getLogger())) {
+                final RecordReader recordReader = recordReaderFactory.createRecordReader(flowFile, in, getLogger())) {
                 final List<String> fieldNames = recordReader.getSchema().getFieldNames();
                 final RecordSet recordSet = recordReader.createRecordSet();
+                final String tableName = context.getProperty(TABLE_NAME).evaluateAttributeExpressions(flowFile).getValue();
+                final KuduTable kuduTable = kuduClient.openTable(tableName);
```

Comment:
How expensive is this operation? With a lot of databases, we need to create the connections in advance because getting one per operation would be prohibitive, so I'd like some information on how Kudu handles this.

Reply:
The connection to Kudu is still acquired in onSchedule (so all the crypto heavy stuff is there), the openTable function basically retrieves table metadata from Kudu server over existing connection.

Observed no visible performance degradation in our environment loading ~60mil records from 600 flow files with 10000 batch size, For lots of small flow files with few rows this change will add a trip to Kudu for metadata retrieval

Reply:
Ok, sounds reasonable.